sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -161,6 +161,7 @@ int pollSocketUntilEventOrTimeout(
     result = poll(&poller, 1, pollTimeoutMs);
     if (result < 0 && EINTR == errno)
     {
+      now = std::chrono::steady_clock::now();
       continue;
```

Comment:
I really don't like this `continue` in the middle of the loop, which ultimately is what caused the `now` update on line 173 to not get hit. I wish we only had to update the `now` local in one place, and we didn't have this `continue` in the middle.

That said, I couldn't come up with a simple refactor (with the platform specific ifdefs) which removes this mid-loop continue, cleanly, with a minimal change while also keeping the existing logic intact.

What you have here is a great minimal change to address the issue.

Reply:
You could do an extra now check on the first line of the loop. Downside, It would mean that now() gets called twice before the first iteration... now() isn't cheap which is why I didn't suggest it.